### DISCUSSION TITLE
Pass number of graphs to dataloader

### DISCUSSION THREAD

**Discussion by dgm2** [pyg-team/pytorch_geometric#4029] [Q&A]
Is it possible to pass as parameter the number of graphs in the datasets for splitting train, valid, test:
for example ZINC instead of 10.000, 1000, 1000 to 1,1,1 for local development?

Thanks!

**Comment by rusty1s** [comment]
You can simply slice the dataset after processing to achieve this, right?
```python
train_dataset = train_dataset[:10]
```

**Comment by dgm2** [comment]
by doing ZINC[:1]  
the slice object inside dataset is still set to the full dataset, do I need to slice each element inside slices?

![image](https://user-images.githubusercontent.com/74187222/153028690-ccddc0a9-610a-411a-8551-d684c22627fb.png)

**Reply by rusty1s** [reply]
Yes, slicing will just internally shrink the dataset (similar to how PyTorch slices Tensors). If you want to create a new dataset only containing the subset of elements, try `ZINC[:1].copy()`.
